src/small_vec.rs

```diff
@@ -0,0 +1,201 @@
+const WORD_SIZE: usize = 64;
+
+#[derive(PartialEq, Debug)]
+pub struct SmallVec {
+    len: usize,
+    curr: usize,
+    bits: Vec<u64>,
+    item_width: usize,
+}
+
+impl Iterator for SmallVec {
+    type Item = u64;
+
+    fn next(&mut self) -> Option<Self::Item> {
+        if self.curr < self.len {
+            let result = Some(self.get(self.curr));
+            self.curr += 1;
+            result
+        } else {
+            None
+        }
+    }
+}
+
+impl SmallVec {
+
+    /// Return the value at the specified `index`
+    /// # Example
+    /// ```
+    /// use compressed_dst::small_vec::SmallVec;
+    /// let v = vec![1, 2, 3, 4];
+    /// let small_vec = SmallVec::from_u64_vec(v);
+    /// assert_eq!(small_vec.get(3), 4);
+    /// ```
+    pub fn get(&self, index: usize) -> u64 {
+        let item_index = (index * self.item_width) / WORD_SIZE;
+        let start = (index * self.item_width) % WORD_SIZE;
+        if start + self.item_width < WORD_SIZE {
+            let mask = ((1 << self.item_width) - 1)
+                << (WORD_SIZE - self.item_width - start);
+            let item = (self.bits[item_index] & mask) >>
+                (WORD_SIZE - self.item_width - start);
+            item
+        } else if self.item_width == WORD_SIZE {
+            self.bits[item_index]
+        } else {
+            let bits_written = WORD_SIZE - start;
+            let mask = ((1 << bits_written) - 1)
+                << (WORD_SIZE - bits_written - start);
+            let first_half = (self.bits[item_index] & mask)
+                >> (WORD_SIZE - bits_written - start);
+            // second half
+            let remaining_bits = self.item_width - bits_written;
+            if remaining_bits > 0 {
+                let mask = ((1 << remaining_bits) - 1)
+                    << (WORD_SIZE - remaining_bits);
+                let second_half = (self.bits[item_index + 1] & mask)
+                    >> (WORD_SIZE - remaining_bits);
+                (first_half << remaining_bits) + second_half
+            } else {
+                first_half
+            }
+        }
+    }
+
+    /// Return a `SmallVec` containing a compressed version of the elements of
+    /// `vec`.
+    pub fn from_u64_vec(vec: Vec<u64>) -> SmallVec {
+        let len = vec.len();
+        let item_width = if let Some(v) = vec.iter().max() {
+            let log = (*v as f64).log2();
+            if log == log.ceil() {
+                (log + 1.0) as usize`
```

Comment:
The stray backtick means this doesn't compile:

```
   Compiling compressed_dst v0.1.0 (file:///home/ltratt/scratch/softdev/gabi/compressed-data-structs)
error: unknown start of token: `
  --> src/small_vec.rs:73:37
   |
73 |                 (log + 1.0) as usize`
   |                                     ^

error: Could not compile `compressed_dst`.

To learn more, run the command again with --verbose.
```

Reply:
Ah, I must've added that by mistake when I was documenting the code. Fixed!